tests/test_hooks.py

```diff
@@ -0,0 +1,70 @@
+import unittest
```

Comment:
Nicer way of testing hooks is actually creating a custom backend with the assertions